source/tutorial.rst

```diff
@@ -23,36 +23,158 @@ Note that depending on your code style during declaration you may choose to omit
 .. code :: javascript
     
     let x = 4;
-    let y : number = 4;
+    let y: number = 4;
+    let x: boolean = false; 
 
 
-Arithmetic
-^^^^^^^^^^^^
+Arithmetic Expressions
+^^^^^^^^^^^^^^^^^^^^^^
 
 Arithmetic expressions also look a lot like in any other programming language (for good reason!). 
 
-.. code:: javascript
+.. code :: javascript
     
     let x = 4.5;
     x = (x * 2) + 3; // x = 12
-  
+
+
+Boolean Expressions
+^^^^^^^^^^^^^^^^^^^
+Boolean expressions for many are quite familiar, as we have taken inspiration from other programming languages.
+
+**Equality**
+
+We have for equality two binary operators: equals ``==``, not equals ``!=``. 
+
+.. code :: javascript
+
+    let x = 5;
+    let y = x == 5; // y = true
+    let z = x != 6; // z = true
+
+**Comparison**
+
+For comparison we have the following binary operators: less than ``<``, greater than ``>``, less than or equal ``<=`` and greater than or equal ``>=``.
+
+.. code :: javascript
     
+    let x = 5;
+    let y = x < 3;  // y = false
+    let z = x > 4;  // z = true
+    let a = x <= 5; // a = true
+    let b = x >= 6; // b = false
+
+**Logical And and Or**
+
+These have been implemented with the following binary operators: logical and ``&&``, logical or ``||``.
+
+.. code :: javascript
+
+    let x = true;
+    let y = x && false; // y = false
+    let z = x || y;     // z = true
+
+**Logical Not**
+
+This is implemented as the standard ``!`` unary operator.
+
+..  code :: javascript
+
+    let x = true;
+    let y = !x;  // y = false
+
+
+**Precedence**
+
+The precedence for the boolean logical operators is as follows:
+
+=========  ============ 
+Operator    Precedence
+---------  ------------
+  ``!``        High 
+  ``&&``       Medium
+  ``||``       Low 
+=========  ============
+
+Examples
+
+===================== ========= ==========================
+ ``A || B && C``        means     ``A || (B && C)``
+``A && B || C && D``    means    ``(A && B) || (C && D)``
+``A && B && C || D``    means    ``((A && B) && C) || D``
+``!A && B || C``        means    ``((!A) && B) || C``
+===================== ========= ==========================
+
 Constructors
 ^^^^^^^^^^^^
 
 Data structures baked into the language have constructors. These can be invoked using the ``new`` keyword.
 
 Note that if a data structure (as below) takes generic type arguments in their constructor they must not be omitted.
 
-.. code:: javascript
+.. code :: javascript
     
     let stack = new Stack<number>;
 
 
 Control structures
 ^^^^^^^^^^^^^^^^^^
 
-``Work in progress!``
+**The if-then and if-then-else Statements**
```

Comment:
Rename to Conditional Statements?

Reply:
You could argue that while and do while are also conditional so I think this is fine. It is how it is in the java documentation

Reply:
Okay sure, we can maybe have a whole conditional statements section later with if and while as subsections